tests/testthat/test-convert_to_na.R

```diff
@@ -83,11 +91,15 @@ test_that("convert_to_na other classes", {
   out <- data.frame(
     a = c(1, 2, NA, 4, 5),
     b = factor(c("a", "b", NA, "d", "e"), levels = letters[1:5]),
-    c = as.Date(c("2022-03-22", NA, "2022-02-02", "2021-04-02", "2020-01-19")),
+    c = as.Date(c("2022-03-22", "2022-01-02", "2022-02-02", "2021-04-02", "2020-01-19")),
     d = c(NA, NA, FALSE, FALSE, NA),
     e = as.complex(1:5)
   )
-  expect_message(x <- convert_to_na(d, na = list(3, "c", TRUE, "2022-01-02")))
+  expect_message(
+    convert_to_na(d, na = list(3, "c", TRUE, "2022-01-02")),
+    "must be of class 'Date'"
+  )
+  expect_message(x <- convert_to_na(d, na = list(3, "c", TRUE, as.Date("2022-01-02"))))
```

Comment:
Maybe use `expect_no_message()` to clarify the expectation?

Reply:
https://testthat.r-lib.org/reference/expect_no_error.html

Reply:
this was actually a mistake, there shouldn't be an `expect_message()` here